core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -312,4 +325,182 @@ private MetricsReporter metricsReporter() {
+      viewVersionBuilder.defaultNamespace(namespace);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperties(Map<String, String> newProperties) {
+      this.properties.putAll(newProperties);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperty(String key, String value) {
+      this.properties.put(key, value);
+      return this;
+    }
+
+    @Override
+    public View create() {
+      ViewOperations ops = newViewOps(identifier);
+      if (null != ops.current()) {
+        throw new AlreadyExistsException("View already exists: %s", identifier);
+      }
+
+      long timestampMillis = System.currentTimeMillis();
+
+      ViewVersion viewVersion =
+          viewVersionBuilder
+              .versionId(1)
+              .addAllRepresentations(viewRepresentations)
+              .timestampMillis(timestampMillis)
+              .putSummary("operation", "create")
+              .build();
+
+      ViewMetadata.Builder builder =
+          ViewMetadata.builder()
+              .setProperties(properties)
+              .setLocation(defaultWarehouseLocation(identifier));
+
+      if (null != schema) {
+        builder.addSchema(schema);
+      }
+
+      ViewMetadata viewMetadata =
+          builder.addVersion(viewVersion).setCurrentVersionId(viewVersion.versionId()).build();
+
+      try {
+        ops.commit(null, viewMetadata);
+      } catch (CommitFailedException ignored) {
+        throw new AlreadyExistsException("View was created concurrently: %s", identifier);
+      }
+
+      return new BaseView(ops, fullTableName(name(), identifier));
+    }
+
+    @Override
+    public View replace() {
+      ViewOperations ops = newViewOps(identifier);
+      if (null == ops.current()) {
+        throw new NoSuchViewException("View does not exist: %s", identifier);
+      }
+
+      ViewMetadata metadata = ops.current();
+
+      long timestampMillis = System.currentTimeMillis();
+
+      ViewVersion viewVersion =
+          viewVersionBuilder
+              .versionId(metadata.currentVersionId() + 1)
+              .addAllRepresentations(viewRepresentations)
+              .timestampMillis(timestampMillis)
+              .putSummary("operation", "replace")
+              .build();
+
+      ViewMetadata.Builder builder = ViewMetadata.buildFrom(metadata).setProperties(properties);
+
+      if (null != schema) {
+        builder.addSchema(schema);
+      }
+
+      ViewMetadata replacement =
+          builder.addVersion(viewVersion).setCurrentVersionId(viewVersion.versionId()).build();
+
+      try {
+        ops.commit(metadata, replacement);
+      } catch (CommitFailedException ignored) {
+        throw new AlreadyExistsException("View was updated concurrently: %s", identifier);
+      }
+
+      return new BaseView(ops, fullTableName(name(), identifier));
+    }
+
+    @Override
+    public View createOrReplace() {
+      if (null == newViewOps(identifier).current()) {
```

Comment:
There should be no need to create a `ViewOperations` multiple times, since the identifier doesn't change.